src/industry_gui.cpp

```diff
@@ -1317,25 +1317,53 @@ class IndustryDirectoryWindow : public Window {
 		static CargoSuffix cargo_suffix[lengthof(i->produced_cargo)];
 		GetAllCargoSuffixes(CARGOSUFFIX_OUT, CST_DIR, i, i->type, indsp, i->produced_cargo, cargo_suffix);
 
-		/* Industry productions */
+		/* Get sorted industry productions */
+		struct Cargo {
```

Comment:
inline struct, that's a bit bold :)
"Cargo" is a bit generically named, are there no name conflicts/shadowing going on?